File: Swift4/Swift4/model/PartijaTriIgraca.swift
//
//  PartijaTriIgraca.swift
//  Swift4
//
//  Created by Tomislav Jakopec on 22/02/2020.
//  Copyright © 2020 Tomislav Jakopec. All rights reserved.
//

import Foundation
class PartijaTriIgraca: Partija{
    
    public override func getRezultat()->Rezultat{
        let rezultat = super.getRezultat()
        rezultat.setTreci(treci: getMjesanja().reduce(0) { $0 + $1.getRezultat().getTreci() })
        return rezultat
    }
    
    
    
    public override var description: String {
        let rezultat = getRezultat();
        return "Partija TRI IGRAČA, igra gotova: " + String(super.isIgraGotova()) + ", " +  super.getIgraci()[0].description + ": " +
                String(rezultat.getPrvi()) +
            " | " + super.getIgraci()[1].description + ": " + String(rezultat.getDrugi()) + " | " + super.getIgraci()[2].description + ": " + String(rezultat.getTreci());
    }
    
    
    
}

File: JS/JS/partija/partijaTriIgraca.js
import { PartijaDvaIgraca } from './partijaDvaIgraca.js'

export class PartijaTriIgraca extends PartijaDvaIgraca {

    constructor(id, doKolikoSeIgra, lokacija, unosi, mjesanja, igraci) {
        super(id, doKolikoSeIgra, lokacija, unosi, mjesanja, igraci)
    }

    toString() {
        const rezultat = this.getRezultat()
        const igraci = this.getIgraci()

        return `Partija TRI IGRAČA, igra gotova: ${this.isIgraGotova()}, ` +
            `${igraci[0]}: ${rezultat.getPrvi()} | ` +
            `${igraci[1]}: ${rezultat.getDrugi()} | ` +
            `${igraci[2]}: ${rezultat.getTreci()}`
    }
}